### PR TITLE
CFY-4691 document fabric plugin native ctx support

### PR DESCRIPTION
**Pull request by nir0s**

### CODE REVIEW

File: content/plugins/fabric.md [hunk 1]
```diff
@@ -156,8 +156,7 @@ node_templates:
 {{< /gsHighlight >}}
 
 {{% gsNote title="Note" %}}
-The fabric plugin doesn't support evaluating Python scripts as the script plugin does and therefore the `ctx` object cannot be used in Python scripts as the script plugin allows.
-Accessing the `ctx` API should be done by calling the `ctx` process. For example: ```os.system('ctx logger info "hello"')```.
+Starting with Cloudify 3.4, the `ctx` object can be used the same way it is used in the script plugin aside from a few minor differences. See [ctx for fabric-plugin](#ctx-for-fabric-plugin) for more info.
```
Comment (dankilman) [inline content/plugins/fabric.md]: mention you are referring python based script. regular ctx process was available previoulsy

File: content/plugins/fabric.md [hunk 2]
```diff
@@ -238,3 +237,47 @@ Using a tasks file instead of a list of commands will allow you to use python co
 Using a list of commands might be a good solution for very simple cases in which you wouldn't want to maintain a tasks file.
 
 {{% /gsTip %}}
+
+
+# ctx for fabric-plugin
+
+Starting with Cloudify 3.4 and fabric-plugin 1.4, we now support using ctx in Python scripts executed by the fabric plugin on remote machines. Most of the functionality is similiar to how the script plugin exposes the ctx object.
```
Comment (dankilman) [inline content/plugins/fabric.md]: ctx -> `ctx`

File: content/plugins/fabric.md [hunk 3]
```diff
@@ -238,3 +237,47 @@ Using a tasks file instead of a list of commands will allow you to use python co
...
+
+## Executing ctx commands
+
+Previously, to use the fabric plugin to execute Python scripts you had to use ctx commands like so:
```
Comment (dankilman) [inline content/plugins/fabric.md]: ctx -> `ctx`

File: content/plugins/fabric.md [hunk 4]
```diff
@@ -238,3 +237,47 @@ Using a tasks file instead of a list of commands will allow you to use python co
...
+
+or 
+
+{{< gsHighlight  python  >}}
+from cloudify import ctx
+
+ctx('logger info Hello!')
+{{< /gsHighlight >}}
+
+The first example shows native ctx usage which can be used to perform most of the trivial actions you can perform using the script plugin like using the logger; retrieving runtime properties and setting them for both node instances and relationship instances; retrieving node properties; downloading blueprint resources; aborting operations, and more.
```
Comment (dankilman) [inline content/plugins/fabric.md]: using the script plugin, like using the logger

File: content/plugins/fabric.md [hunk 5]
```diff
@@ -238,3 +237,47 @@ Using a tasks file instead of a list of commands will allow you to use python co
...
+The most notable difference is that to get all properties or runtime properties for a node you would have to do the following:
+
+{{< gsHighlight  python  >}}
+from cloudify import ctx
+
+my_node_properties = ctx.node.properties.get_all()
+my_instance_runtime_properties = ctx.instance.runtime_properties.get_all()
+{{< /gsHighlight >}}
+
+This is also true for `source` and `target` runtime properties.
```
Comment (dankilman) [inline content/plugins/fabric.md]: `source` and `target` properties and runtime properties

File: content/plugins/fabric.md (hunk 2 above)
Comment (dankilman) [inline content/plugins/fabric.md]: i don't think the word "we" belongs here. Perhaps: "Cloudify now supports", "The fabric plugin now supports", etc...

File: content/plugins/fabric.md (hunk 4 above)
Comment (dankilman) [inline content/plugins/fabric.md]: you don't set runtime properties of relationship instances but rather you set the source/target node instances runtime properties in relationship operations

File: content/plugins/fabric.md [hunk 6]
```diff
@@ -238,3 +237,47 @@ Using a tasks file instead of a list of commands will allow you to use python co
...
+
+or 
+
+{{< gsHighlight  python  >}}
+from cloudify import ctx
+
+ctx('logger info Hello!')
+{{< /gsHighlight >}}
+
+The first example shows native ctx usage which can be used to perform most of the trivial actions you can perform using the script plugin, like using the logger; retrieving runtime properties and setting them for node instances; setting the source/target node instances runtime properties in relationship operations; retrieving node properties; downloading blueprint resources; aborting operations, and more.
```
Comment (dankilman) [inline content/plugins/fabric.md]: i kinda assumed all ctx references will be changes to `ctx`

Comment (nir0s) [inline content/plugins/fabric.md]: woops

File: content/plugins/fabric.md [hunk 7]
```diff
@@ -238,3 +237,47 @@ Using a tasks file instead of a list of commands will allow you to use python co
...
+from cloudify import ctx
+
+ctx('logger info Hello!')
+{{< /gsHighlight >}}
+
+The first example shows native ctx usage which can be used to perform most of the trivial actions you can perform using the script plugin, like using the logger; retrieving runtime properties and setting them for node instances; setting the source/target node instances runtime properties in relationship operations; retrieving node properties; downloading blueprint resources; aborting operations, and more. 
+
+The second example shows that you can still use ctx to execute commands as if you're running it from a bash script.
+
+The most notable difference is that to get all properties or runtime properties for a node you would have to do the following:
```
Comment (nir0s) [inline content/plugins/fabric.md]: node and node instance respectively

File: content/plugins/fabric.md [hunk 8]
```diff
@@ -238,3 +237,47 @@ Using a tasks file instead of a list of commands will allow you to use python co
...
+
+{{< gsHighlight  python  >}}
+from cloudify import ctx
+
+my_node_properties = ctx.node.properties.get_all()
+my_instance_runtime_properties = ctx.instance.runtime_properties.get_all()
+{{< /gsHighlight >}}
+
+This is also true for `source` and `target` node properties and node instance runtime properties.
```
Comment (dankilman) [inline content/plugins/fabric.md]: This is also true -> The same holds true for

**Pull request merged**